src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -548,19 +547,19 @@ private ObjectSpec CreateObjectSpec(TypeParseInfo typeParseInfo)
                     if (!hasPublicParameterlessCtor && hasMultipleParameterizedCtors)
                     {
                         initDiagDescriptor = DiagnosticDescriptors.MultipleParameterizedConstructors;
-                        initExceptionMessage = string.Format(Emitter.ExceptionMessages.MultipleParameterizedConstructors, typeName);
+                        initExceptionMessage = string.Format(Emitter.ExceptionMessages.MultipleParameterizedConstructors, typeSymbol.GetFullName());
```

Comment:
>typeSymbol.GetFullName() [](http://example.com/codeflow?start=122&length=24)

Looks this can be called multiple times. do we need to cache it better?

Reply:
The string is only used when we emit diagnostics. Given that diagnostics tend to be exceptional, I intentionally changed it so that we don't have to create the string every time.